### PR TITLE
[INLONG-7004][Manager] Support to creating schema of StreamSource by JSON

### LINKED ISSUE #7004: [Feature][Dashboard][Manager] Creating schema of  StreamSource by Statement
### Description

At present, when creating a Stream field, it needs to be input one by one, which is relatively inefficient, especially when there are too many fields, the input will be very difficult.

image

image

In order to improve the efficiency of creating new Stream fields, the following three batch import methods are provided:
1. CSV: "Comma-Separated Values", is a file format for storing and exchanging data in a tabular structure, where each row represents a record and each column represents a field. The values in each row are separated by commas, and each row is separated by a new line.
2. Standard SQL: Specify the field name, field type and field remarks through DDL statements
3. Excel: import Excel template
4. JSON：Input a JSON statement and specify the information of the field through three keys: Name, Type, and Comment. 

image

we could complete this feature refer to:
image

- [x] #7415  
- [x] #7688  
- [x] #7720 
- [x] #7690 
- [x] #7789

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #7415: [Feature][Dashboard][Manager] Creating schema of StreamSource by JSON
### Description

we could complete this feature refer to:
image

### Use case

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #7415 

### Motivation

Support to creating schema of StreamSource and StreamSink by JSON

### Modifications

Support to creating schema of StreamSource and StreamSink by JSON

### CODE REVIEW

**Comment by dockerzhang** [comment]
@fuweng11 please add related UTs.

**Comment by fuweng11** [comment]
> @fuweng11 please add related UTs.

OK

**Comment by gong** [comment]
@fuweng11 Please fix UT error
```
 SortServiceImplTest.prepareAll:250->prepareCluster:342 » DuplicateKey 
### Error updating database.  Cause: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: Unique index or primary key violation: "PUBLIC.UNIQUE_INLONG_CLUSTER_INDEX_C ON PUBLIC.INLONG_CLUSTER(NAME NULLS FIRST, TYPE NULLS FIRST, IS_DELETED NULLS FIRST) VALUES ( /* key:51 */ CAST('test-cluster-1' AS VARCHAR_IGNORECASE(14)), CAST('HIVE' AS VARCHAR_IGNORECASE(4)), 0)"; SQL statement:
insert into inlong_cluster (id, name, type,
                                    url, cluster_tags, ext_tag,
                                    token, ext_params, description,
                                    heartbeat, in_charges,
                                    status, creator, modifier)
        values (?, ?, ?,
                ?, ?, ?,
                ?, ?, ?,
                ?, ?,
                ?, ?, ?) [23505-214]
```

**Review by gong — APPROVED** [review]

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongStreamController.java [hunk 1]
```diff
@@ -169,4 +176,10 @@ public Response<Boolean> delete(@RequestParam String groupId, @RequestParam Stri
         return Response.success(streamService.delete(groupId, streamId, username));
     }
 
+    @RequestMapping(value = "/stream/parseFields", method = RequestMethod.POST)
+    @ApiOperation(value = "parse inlong stream fields")
```
Comment (healchow) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongStreamController.java:180]: 1. "parse inlong stream fields" -> "Parse inlong stream fields from JSON string"
2. Add `@ApiImplicitParam` for `fieldsJson`.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkService.java [hunk 2]
```diff
@@ -245,4 +245,8 @@ public interface StreamSinkService {
      */
     Boolean updateAfterApprove(List<SinkApproveDTO> sinkApproveList, String operator);
 
+    /**
+     * Converts a json string to a sinkFields
+     */
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkService.java:250]: Please add `@param` and `@return` for public methods.

**Review by healchow — CHANGES_REQUESTED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Review by vernedeng — APPROVED** [review]
LGTM

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java [hunk 3]
```diff
@@ -720,6 +725,24 @@ public void logicDeleteDlqOrRlq(String groupId, String topicName, String operato
...
+                    });
+            return fieldsMap.keySet().stream().map(fieldName -> {
+                StreamField field = new StreamField();
+                field.setFieldName(fieldName);
+                field.setFieldType(fieldsMap.get(fieldName));
+                return field;
+            }).collect(Collectors.toList());
+        } catch (Exception e) {
+            LOGGER.error("parse inlong stream fields error", e);
+            throw new JsonException("parse inlong stream fields error");
```
Comment (gosonzhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java:742]: The corresponding error code needs to be added

**Review by gosonzhang — APPROVED** [review]

**Pull request merged**